Framework/API/src/Algorithm.cpp

```diff
@@ -1622,6 +1628,26 @@ bool Algorithm::getAlgStartupLogging() const {
   return m_isAlgStartupLoggingEnabled;
 }
 
+void Algorithm::declareProperty(std::unique_ptr<Kernel::Property> p,
+                                const std::string &doc) {
+  if (isReserved(m_reservedList, p->name()))
```

Comment:
I think this is not sufficient, you also need to check the `WorkspaceIndexType` and `WorkspaceIndexSet` names?